minio/time.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# MinIO Python Library for Amazon S3 Compatible Cloud Storage, (C)
+# 2020 MinIO, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Time formatter for S3 APIs."""
+
+from __future__ import absolute_import
+
+import locale
+from contextlib import contextmanager
+from datetime import datetime, timezone
+
+from . import __LOCALE_LOCK__
+
+_HTTP_HEADER_FORMAT = "%a, %d %b %Y %H:%M:%S GMT"
+
+
+@contextmanager
+def _set_locale(name):
+    """Thread-safe wrapper to locale.setlocale()."""
+    with __LOCALE_LOCK__:
```

Comment:
Is this a replacement for `acquire() and release()`?

Reply:
Refer https://stackoverflow.com/questions/10525185/python-threading-how-do-i-lock-a-thread